src/Compilers/Core/Desktop/CodePage1252Encoding.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.Text
+{
+    /// <summary>
+    /// CodePage 1252 decoder. This is a single byte character set encoding that is very close
+    /// to Latin1 except some of the code points in the control set 1 range (0x80 to 0x9F) are
+    /// replaced with typographic characters.
+    /// </summary>
+    internal class CodePage1252Encoding : Encoding
+    {
+        public static readonly CodePage1252Encoding Instance = new CodePage1252Encoding();
+
+        public override int CodePage => 1252;
+
+        public override int GetByteCount(char[] chars, int index, int count)
+        {
+            // This is a decoder only
```

Comment:
You mean "CodePage1252Decoder"? I think that might be a bit incongruous since it "is-a" Encoding and we might, indeed, need to fill in the encoder parts at a later date.